src/network/network_client.cpp

```diff
@@ -1173,7 +1176,6 @@ void ClientNetworkGameSocketHandler::CheckConnection()
 	if (lag > 20) {
 		this->NetworkGameSocketHandler::CloseConnection();
 		ShowErrorMessage(STR_NETWORK_ERROR_LOSTCONNECTION, INVALID_STRING_ID, WL_CRITICAL);
```

Comment:
ShowErrorMessage is repeated inside CloseConnection

Reply:
That's not the subject of this PR :)
And it definitely does not hurt performance as much as saving twice does.